src/app/containers/WalletProvider/saga.ts

```diff
@@ -38,6 +38,12 @@ function createBlockPollChannel({ interval, web3 }) {
       }
     });
 
+    blockTracker.on('error', () => {
+      // Restart block tracker
+      blockTracker.removeAllListeners('');
+      emit(actions.connectionError());
```

Comment:
The flow is like this:
- SovrynNetwork tries to connect to first rpc
- blockTracker gets initalised with the currentProvider, which is the first rpc
- New SovrynNetwork retry logic triggers, second rpc works
- blockTrackers provider instance is still the first rpc, which fails now, emits error
- connectionError action will trigger calling createBlockPollChannel again
- parameter web3 is the same _readWeb3, but with the (in the meantime) updated provider (currentProvider)
- new blockTracker can connect to this one, as it was already successful

I tried this with changing mainnet.sovryn.app with a non-existing domain and it worked (after the first spam of course).

Reply:
Yes on refresh your solution works, it will correctly cycle through the fallbacks until one sticks, but if the RPC is suddenly unavailable after successful initialization of the SovrynNetwork it will stay on that RPC and spam connectionError redux actions until the connection comes back. This leads to heavily degraded performance of the frontend, which is why I suggested atleast throttling the connectionError retries so that the reconnect only happens every few seconds. 
If we could detect a node failure at the SovrynNetwork level though that would probably be the ideal place to trigger a switch to another rpc.

Reply:
Ahh, I understand. Will try my best and (hopefully) come with a solution til tuesday.

Reply:
Thank you, if you don't find time or a solution, let us know and a team member will take a look.

Reply:
I don't know much about redux or saga, but I think the solution is not that bad. It works and the spamming got better. Every rpc change now triggers a block-tracker recreate, and if the block-tracker errors, we try to reconnect too after a cooldown.

Reply:
@ShynRou Sorry to bother, just making sure I understand this GitHub CodeReview-thingy: You get a mail saying that I changed something and I don't have to resolve the conversation for that (that's the reviewers part I guess), right?

Reply:
Yeah, I get a mail when you update something. Usually you re-request review at the top, and that is the sign that the reviewer should take another look. If it's a small issue you fixed or you just applied a suggestion you can resolve it yourself, otherwise let the reviewer do it.

Sadly I don't have time to re-test this right now, I will try to get back to you on monday. Thank you for your help :slightly_smiling_face:

Reply:
Very good! Yeah, did the re-request review last time, option does not show up on last change, so I was unsure.

No worries, no stress.

You're welcome. I am happy to be able to enhance Sovryn!